### ISSUE TITLE
CLS add ~20 second to game load

### ISSUE THREAD

**Issue by Krutoy242** [AlexIIL/BetterLoadingScreen#74]
With CLS [my modpack](https://github.com/Krutoy242/Enigmatica2Expert-Extended) loads about 20 seconds slower.

I have small tool that fetch `debug.log` and gather all `[...] took 0.001s` values.

**Without** CLS there is more than 200 mods that loads less then `0.1s`.

Total Load Time | Mods count | Description
---|---|:---
76.52s | 43 | Other mods;
51.77s | 156 | 'Fast' mods (load 1.0s - 0.1s);
 6.76s | 216 | 'Instant' mods (load %3C 0.1s)

**With** CLS mostly all of this "instant" mods increased their load time for 1.0s - 0.1s, so whole game load time increased for about 25s.

Total Load Time | Mods count | Description
---|---|:---
77.15s |  44 |  Other mods;
86.83s |  334  | 'Fast' mods (load 1.0s - 0.1s);
 2.14s |  38 |  'Instant' mods (load %3C 0.1s)

This how my loading screen setup looks like:
![javaw_QQQxiZzE0Z](https://user-images.githubusercontent.com/936618/145375082-aeb0b18f-3b90-4ef2-b957-2d441e7c5530.png)

I don't know how CLS working, but it feels like it drawing screen **in sync mode**, that cause little lag for frame rendering for each line of status log, so multiplying to 200 it adds this lag.

How to fix that? Fancy loading screen should not cost 20 seconds!

**Comment by AlexIIL** [comment]
I think this might be caused by the CLS config option `smooth_init` defaulting to `true` rather than `false` - can you check to see if changing it does anything?

**Comment by Krutoy242** [comment]
Right now my config option set to `false`:
```js
    B:smooth_init=false
    B:use_custom=true
    B:use_frame=false
```

Should i enable it to `true`?

**Comment by AlexIIL** [comment]
No, I was hoping smooth init was the cause of this. Setting it to true would probably make it much slower then.

**Comment by AlexIIL** [comment]
I'll try testing it locally - have you made any changes to the modpack (other than adding CLS) from what you're testing ATM and release 0.41.1?

**Comment by Krutoy242** [comment]
> have you made any changes

Yes, i did, but i didn't pushed them yet. I would make new release in next few hours.

**Comment by AlexIIL** [comment]
Ok, from some local testing:

 - Forge (default): 8:18
 - Forge (disabled): 7:23
 - CLS (smooth_init=true): 9:39
 - CS (smooth_init=false): 8:46

So yes, CLS seems to make it slower. I'd have to start debugging a bit deeper to find out why though. `smooth_init` intentionally sleeps so it's always going to make it slower, but every frame of the loading screen has to sync anyway - even with forge - so I'm not sure how much I can speed this up by.

**Comment by Krutoy242** [comment]
I would try to "play with options too if this would help.

1. I tried to load with `I:fps_limit=10` and nothing changes.
2. I also though there is problem with `time` getter, but even removing it, nothing changes, still slow.
3. Set `I:fps_limit=300` wont change load time too
